application/libraries/MenuObjects/Menu.php

```diff
@@ -131,4 +147,20 @@ public function getTooltip()
     {
         return $this->tooltip;
     }
+
+    /**
+     * @return boolean
+     */
+    public function isPrepended()
+    {
+        return $this->isPrepended;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isButton()
+    {
+        return $this->isButton;
```

Comment:
You can return false here and true in the subclass, then you don't need a property.

Or use instanceof, but I see in the code that I didn't do that either.